crates/lib/src/api.rs

```diff
@@ -313,3 +315,53 @@ pub async fn retrieve_results(
         .map(ExecutionResults::from)
         .ok_or_else(|| GrpcClientError::ResponseEmpty("Controller Job Execution Results".into()))
 }
+
+/// API Errors encountered when trying to list available quantum processors.
+#[derive(Debug, thiserror::Error)]
+pub enum ListQuantumProcessorsError {
+    /// Failed the http call
+    #[error("Failed to list processors via API: {0}")]
+    ApiError(#[from] OpenAPIError<quantum_processors_api::ListQuantumProcessorsError>),
+
+    /// Pagination did not finish before timeout
+    #[error("API pagination did not finish before timeout: {0:?}")]
+    TimeoutError(#[from] Elapsed),
+}
+
+/// Query the QCS API for the names of all available quantum processors.
+/// If `None`, the default `timeout` used is 10 seconds.
+pub async fn list_quantum_processors(
+    client: &Qcs,
+    timeout: Option<Duration>,
+) -> Result<Vec<String>, ListQuantumProcessorsError> {
+    let timeout = timeout.unwrap_or_else(|| Duration::from_secs(10));
```

Comment:
I imagine we will, though I'm not sure the timeout value will be the same everywhere - if it is, it should probably live on `qpu::client::Qcs`

Reply:
Though I think it would make sense to add that to this MR, I addressed this comment but split the larger task of api defaults into a different issue in the interest of not letting this MR continue to balloon. https://github.com/rigetti/qcs-sdk-rust/issues/239